src/Microsoft.AspNet.Mvc.ViewFeatures/Rendering/Html/DefaultDisplayTemplates.cs

```diff
@@ -286,12 +287,13 @@ private static bool ShouldShow(ModelExplorer modelExplorer, TemplateInfo templat
                 !templateInfo.Visited(modelExplorer);
         }
 
-        public static string StringTemplate(IHtmlHelper htmlHelper)
+        public static IHtmlContent StringTemplate(IHtmlHelper htmlHelper)
         {
-            return htmlHelper.Encode(htmlHelper.ViewData.TemplateInfo.FormattedModelValue);
+            var value = htmlHelper.ViewData.TemplateInfo.FormattedModelValue;
+            return new StringHtmlContent(value == null ? string.Empty : value.ToString());
```

Comment:
I suggest removing the `StringHtmlContent` option to treat content as already-encoded. It's a well-known MVC idiom that `HtmlString` values do not need further HTML encoding.